test-cloud/test-cloud-template/src/test/java/org/kie/cloud/tests/template/TemplateParameterTest.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2018 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+package org.kie.cloud.tests.template;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import io.fabric8.openshift.api.model.Template;
+import io.fabric8.openshift.client.DefaultOpenShiftClient;
+import io.fabric8.openshift.client.OpenShiftClient;
+import io.fabric8.openshift.client.OpenShiftConfig;
+import io.fabric8.openshift.client.OpenShiftConfigBuilder;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.kie.cloud.api.scenario.MissingResourceException;
+import org.kie.cloud.openshift.template.OpenShiftTemplate;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class TemplateParameterTest {
+
+    private static final Logger log = LoggerFactory.getLogger(TemplateParameterTest.class);
+    private static final List<OpenShiftTemplate> availableOpenShiftTemplates = getAvailableOpenShiftTemplates();
+
+    private OpenShiftClient openShiftClient;
+
+    @BeforeClass
+    public static void printTestedTemplates() {
+        log.info("Tested templates:");
+        for (OpenShiftTemplate openShiftTemplate : availableOpenShiftTemplates) {
+            log.info(openShiftTemplate.getTemplateUrl().toString());
+        }
+    }
+
+    @Before
+    public void createOfflineOpenShiftClient() {
+        OpenShiftConfig openShiftConfig = new OpenShiftConfigBuilder()
+                .withDisableApiGroupCheck(true)
+                .build();
+
+        openShiftClient = new DefaultOpenShiftClient(openShiftConfig);
+    }
+
+    @Test
+    public void testTemplateParameterDuplicities() {
+        for (OpenShiftTemplate openShiftTemplate : availableOpenShiftTemplates) {
```

Comment:
I like this PR!  But instead of using for loop in these tests, we should be probably using `@Parametrized` test. In that way if there is problem introduced in 2 templates, we'll see separate failure for each of them, while this approach will only show one failure. WDYT?